scipy/ndimage/src/_rank_filter_1d.cpp

```diff
@@ -0,0 +1,304 @@
+/*
+Started working on https://ideone.com/8VVEa (Copyright (c) 2011 ashelly.myopenid.com
+under <http://www.opensource.org/licenses/mit-license>),
+I optimized by restriction of cases and proper initialization,
+also adapted for rank filter rather than the original median filter.
+Allowed different boundary conditions.
+Moved to C++ for polymorphism and added C-Numpy API.
```

Comment:
Could you remove the comments on evolution of the code, and copy exactly the copyright comment that was in the original file?

Reply:
is it ok with you? I would not insist on the second line but some major changes have been made in did.

//Copyright (c) 2011 ashelly.myopenid.com under <http://www.opensource.org/licenses/mit-license>
//Modified in 2024 by Gideon Genadi Kogan

Reply:
Yes, that is fine with me.